src/main/java/com/example/planit/engine/CalendarEngine.java

```diff
@@ -1320,20 +1320,67 @@ public static boolean hasAuthorizeScopesStillValid(String access_token) throws I
 
             if (maybeUser.isEmpty()) {
-                return new DTOstudyPlanResponseToController(false, ERROR_USER_NOT_FOUND, HttpStatus.BAD_REQUEST);
+                return new DTOstudyPlanAndSessionResponseToController(false, ERROR_USER_NOT_FOUND, HttpStatus.BAD_REQUEST);
             }
-            return new DTOstudyPlanResponseToController(true, NO_PROBLEM, HttpStatus.OK, maybeUser.get().getLatestStudyPlan());
+            User user = maybeUser.get();
+            // refresh access_token before making api call
+            validateAccessTokenExpireTime(user);
+
+            // check if access_token still have scope for Google calendar
+            if (!hasAuthorizeScopesStillValid(user.getAuth().getAccessToken())) {
+                return new DTOstudyPlanAndSessionResponseToController(false, Constants.ERROR_NO_VALID_ACCESS_TOKEN, HttpStatus.UNAUTHORIZED);
+            }
+
+            return new DTOstudyPlanAndSessionResponseToController(true, NO_PROBLEM, HttpStatus.OK, maybeUser.get().getLatestStudyPlan(), getUpComingSessionForUser(user));
         } catch (Exception e) {
             logger.error(buildExceptionMessage(e));
-            return new DTOstudyPlanResponseToController(false, ERROR_DEFAULT, HttpStatus.INTERNAL_SERVER_ERROR);
+            return new DTOstudyPlanAndSessionResponseToController(false, ERROR_DEFAULT, HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
 
+    private StudySession getUpComingSessionForUser(User user) throws GeneralSecurityException, IOException {
+
+        Calendar calendarService = getCalendarService(user.getAuth().getAccessToken(), user.getAuth().getExpireTimeInMilliseconds());
+
+        List<CalendarListEntry> calendarList = getCalendarList(calendarService);
+
+        StudySession upComingSession = null;
+
+        Instant currentTime = Instant.now();
+        DateTime startTime = new DateTime(currentTime.toEpochMilli());
+
+        Events events = null;
+
+        for (CalendarListEntry calendar : calendarList) {
+            if (calendar.getId().equals(user.getPlanItCalendarID())) {
+                events = calendarService.events().list(calendar.getId())
+                        .setTimeMin(startTime)
+                        .setOrderBy("startTime")
+                        .setSingleEvents(true)
+                        .execute();
+                break;
+            }
+        }
+
+        // if the user have calendar. we return the upComing study session.
+        if (events != null) {
+            List<Event> ListOfEvent = events.getItems();
+            if (ListOfEvent.size() != 0)
+            {
+                Event event = ListOfEvent.get(0);
+                upComingSession = new StudySession(event.getStart().getDateTime(), event.getEnd().getDateTime());
+                upComingSession.setCourseName(event.getSummary());
+                upComingSession.setDescription(event.getDescription());
+            }
```

Comment:
@oshriFadlon1 The variable `ListOfEvent` need to have a lower case "L" at the beginning and "s" in the end (so it will be "listOfEvents"). 

Also, maybe change the name of `event` to something more meaningful (upcomingEvent).

Also, it seems odd that you create a new `StudySession` with some arguments and then right after use `setCourseName` and `setDescription`. I suggest you create a ctor of `StudySession` with all the parameters. What do you think? @Danielsio  @oshriFadlon1

Reply:
Agread.

Reply:
i added the ctor to study session and change ListOfEvent to listOfEvents